domain/agentbinary/state/state.go

```diff
@@ -28,6 +32,60 @@ func NewState(factory database.TxnRunnerFactory) *State {
 // It returns [coreerrors.NotSupported] if the architecture is not found in the database.
 // It returns [coreerrors.NotFound] if object store UUID is not found in the database.
 func (s *State) Add(ctx context.Context, metadata agentbinary.Metadata) error {
-	// TODO: implement this method.
-	return nil
+	// Prepare the statements we'll need
```

Comment:
You can define the name and id in architectureRecord, then use the same struct for both input and output.
There are many examples in domain state.
So the SQL could be 
```
SELECT id AS &architectureRecord.id
FROM   architecture
WHERE  name = $architectureRecord.name
```